pin/pin_test.go

```diff
@@ -85,10 +84,6 @@ func TestPinnerBasic(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	if !p.IsPinned(ck) {
-		t.Fatal("Child of recursively pinned node not found")
-	}
-
```

Comment:
why is this test case changing? it shouldn't. `ck` should continue to be considered "pinned".

indirect pins as a _concept_ have not changed. the thing is pinned _indirectly_ and the users should be able to discern that. otherwise the UX gets confusing.